Description: Patch adds keys '1', '3' in replay mode to move marks by +/-5secs.
 When in replay mode key '0' sets a mark and stops there.
 'Green' and 'Yellow' act as 'Prev' and 'Next'.
Author: FrankJepsen at vdrportal.de
Origin: http://vdrportal.de/board/thread.php?postid=718583#post718583

Index: vdr/menu.c
===================================================================
--- vdr.orig/menu.c	2013-04-02 21:42:30.000000000 +0200
+++ vdr/menu.c	2013-04-02 21:45:11.000000000 +0200
@@ -4939,10 +4939,46 @@
                    Stop();
                    return osEnd;
     default: {
+      bool play, forward;
+      int speed;
+      int Current, Total;
+      cMark *m;
+
       DoShowMode = false;
       switch (int(Key)) {
         // Editing:
-        case kMarkToggle:      MarkToggle(); break;
+        case kMarkToggle:      if (GetReplayMode(play, forward, speed) && play) {
+                                  displayFrames = true;
+                                  Pause();
+                                  }
+                               MarkToggle();
+                               break;
+        case k1|k_Repeat:
+        case k1:               if (GetReplayMode(play, forward, speed) && !play && GetIndex(Current, Total) && (m = marks.Get(Current)) != NULL) {
+                                  displayFrames = true;
+                                  int p = SkipFrames(-SecondsToFrames(5, FramesPerSecond()));
+                                  cMark *m2;
+                                  if ((m2 = marks.Prev(m)) != NULL && m2->Position() >= p)
+                                     break;
+                                  m->SetPosition(p); Goto(p, true);
+                                  marks.Save();
+                                  }
+                               else
+                                  SkipSeconds(-20);
+                               break;
+        case k3|k_Repeat:
+        case k3:               if (GetReplayMode(play, forward, speed) && !play && GetIndex(Current, Total) && (m = marks.Get(Current)) != NULL) {
+                                  displayFrames = true;
+                                  int p = SkipFrames(SecondsToFrames(5, FramesPerSecond()));
+                                  cMark *m2;
+                                  if ((m2 = marks.Next(m)) != NULL && m2->Position() <= p)
+                                     break;
+                                  m->SetPosition(p); Goto(p, true);
+                                  marks.Save();
+                                  }
+                               else
+                                  SkipSeconds(20);
+                               break;
         case kPrev|k_Repeat:
         case kPrev:
         case kMarkJumpBack|k_Repeat:
